photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala

```diff
@@ -431,4 +459,74 @@ object ModelProcessingUtils {
     val colLatentFactors = loadLatentFactorsFromHDFS(colLatentFactorsPath.toString, sparkContext)
     new MatrixFactorizationModel(rowEffectType, colEffectType, rowLatentFactors, colLatentFactors)
   }
+
+  /**
+   * Save model metadata to a JSON file.
+   *
+   * @param outputDir The HDFS directory that will contain the metadata file
+   * @param params The model parameters, from which metadata is extracted
+   * @param sparkContext The Spark context
+   */
+  def saveGameModelMetadataToHDFS(outputDir: String, params: Params, sparkContext: SparkContext): Unit = {
+
+    // 2 ancillaries that decompose the problem into more manageable tasks
+    def writeConfigToJson(configs: Map[String, GLMOptimizationConfiguration]) =
+      configs
+        .map { case (cfgName, cfg) => s"""{ "name": "$cfgName", "configuration": ${cfg.toJson} }""" }
+        .mkString(",")
+
+    def writeConfigsArrayToJson(which: String, configsArray: Array[Predef.Map[String, GLMOptimizationConfiguration]]) =
+      s"""
+         |{ "configurations": "$which",
+         |  "values": [
+         |    ${configsArray.map(cfgs => writeConfigToJson(cfgs)).mkString(",")}
+         |  ]
+         |}
+       """.stripMargin
+
+    val feConfigurations = writeConfigsArrayToJson("fixed-effect", params.fixedEffectOptimizationConfigurations)
+    val reConfigurations = writeConfigsArrayToJson("random-effect", params.randomEffectOptimizationConfigurations)
+
+    IOUtils.toHDFSFile(sparkContext, outputDir + "model-metadata.json") {
+      writer => writer.println(
+        s"""
+           |{ "modelType": "${params.taskType}",
+           |  "modelName": "${params.applicationName}",
+           |  "modelVersion": "1.0",
```

Comment:
It's available in one of the systems for which I'm building a converter now, but we don't have it on the photon side yet. I will take it out for now.